src/server.rs

```diff
@@ -266,14 +264,44 @@ impl<C: CommandCreatorSync> SccacheServer<C> {
     fn _run<'a>(self, shutdown: Box<Future<Item = (), Error = ()> + 'a>)
                 -> io::Result<()>
     {
-        let SccacheServer { mut core, listener, rx, service, timeout, wait } = self;
+        let SccacheServer { mut runtime, listener, rx, service, timeout, wait } = self;
 
         // Create our "server future" which will simply handle all incoming
         // connections in separate tasks.
-        let handle = core.handle();
-        let server = listener.incoming().for_each(move |(socket, _addr)| {
+        let server = listener.incoming().for_each(move |socket| {
             trace!("incoming connection");
-            SccacheProto.bind_server(&handle, socket, service.clone());
```

Comment:
Forgot I need to cleanup this part. `tokio-proto` is unmaintained so I had to string together a minimal version of it instead but it still uses some of tokio-proto's types which makes it a bit of a mess.

https://github.com/tokio-rs/tokio-proto.